src/main/scala/GuessWho/Interface.scala

```diff
@@ -0,0 +1,159 @@
+package GuessWho
+
+import scala.io.StdIn.readLine
+
+class Interface {
+  private var _hairQuestions:Map[Int, String] = Map(
+    1 -> "Do they have hair?",
+    2 -> "Do they have blonde hair?",
+    3 -> "Do they have brunette hair?",
+    4 -> "Do they have red hair?",
+    0 -> "Go back"
+  )
+
+  private var _eyeQuestions:Map[Int, String] = Map(
+    1 -> "Do they have blue eyes?",
+    2 -> "Do they have brown eyes?",
+    3 -> "Do they have green eyes?",
+    0 -> "Go back"
+  )
+
+  private var _genderQuestions:Map[Int, String] = Map(
+    1 -> "Are the male?",
+    2 -> "Are the female?",
+    0 -> "Go back"
+  )
+
+  private var _facialQuestions:Map[Int, String] = Map(
+    1 -> "Do they have facial hair?",
+    0 -> "Go back"
+  )
+
+  private var _glassesQuestions:Map[Int, String] = Map(
+    1 -> "Do they have glasses?",
+    0 -> "Go back"
+  )
+
+  private var _hatQuestions:Map[Int, String] = Map(
+    1 -> "Do they have a hat?",
+    0 -> "Go back"
+  )
+
+
+  def displayCharacters(characters:Seq[Character]):Unit = {
+    characters.foreach { character =>
+      println(
+        s"""
+           |Name: ${character.name}
+           |Has Hair: ${character.hasHair}
+           |Has Facial Hair: ${character.hasFacialHair}
+           |Has Glasses: ${character.hasGlasses}
+           |Has Hat: ${character.hasHat}
+           |Gender: ${character.gender}
+           |Eye Colour: ${character.eyeColour}
+           |Hair Colour: ${character.hairColour}
+    """.stripMargin
+      )
+    }
+    println(s"Number of remaining characters: ${characters.length}")
+  }
+
+  def get_user_input(message: String): String = {
+    print(message)
+    val input = readLine()
+    input
+  }
+
+  def validate_attribute_choice(choice: String): Int = {
+    try{
+      choice.toInt
+    } catch{
+      case e: NumberFormatException => {
```

Comment:
Was going back and forth looking for where we actually take the choice and convert it to an Int, and its here!

This is the first place I would try and change

Even as simple as matching off the `Int`s here then returning a `Choice` case class?

For now throwing this exception is fine, but an `Either` would make it clear that when we consume this method it could lead to an error!

Side note, the converting from `Int` to `Choice` should be in the `Choice` case class' companion object